doc/nrf/device_guides/working_with_nrf/nrf70/developing/fw_patches_ext_flash.rst

```diff
@@ -0,0 +1,133 @@
+.. _ug_nrf70_developing_fw_patch_ext_flash:
+
+Firmware patches in the external flash
+######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Overview
+********
+
+The nRF70 Series firmware patch is memory intensive and is by default part of the internal flash.
+In order to save flash memory for resource constrained hosts, the nRF70 Series offers a feature to move firmware patches to the external flash, if the external flash is available.
+
+Pre-requisites
+--------------
+
+The following conditions must be met to use this feature:
+
+- The external flash must be available and configured in the device tree.
+- The external flash must be large enough to store the firmware patches, maximum size is 128 KB.
+
+Constraints
+-----------
+
+The following constraints apply to this feature:
+
+- Access to the external flash is only supported via the QSPI interface for now.
+- In nRF SoC's XIP feature is only available via the QSPI interface only.
+
+
+Supported platforms
+-------------------
+
+The following platforms are supported:
+
+- nRF5340 SoC or DK
+- nRF52840 SoC or DK
+
+.. note:: Due to constraints mentioned above, the nRF7002 SoC or DK is not supported as it needs access to external flash via the SPI interface.
+
+Details
+*******
+
+There are two ways to move the firmware patch to the external flash:
+
+* XIP Supported external flash
+* non-XIP Supported external flash
+
+
+XIP Supported external flash
+----------------------------
+If the external flash supports XIP, then the firmware patch can be relocated to the external flash and executed from there.
+To enable this feature, set the following configuration option: :kconfig:option:`CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_XIP=y`.
+Once the build is complete, the feature can be verified by checking the memory regions summary in the build output.
+A new memory region called ``EXTFLASH`` is added to the memory regions summary, and the firmware patch is placed in this region.
+And the ``FLASH`` used size is reduced by the size of the firmware patch.
+
+A sample memory regions summary is shown below:
+
+.. code-block:: console
+
+  Memory region         Used Size  Region Size  %age Used
+        EXTFLASH:       64544 B         8 MB      0.77%
+           FLASH:      618752 B         1 MB     59.01%
+             RAM:      170636 B       448 KB     37.20%
+        IDT_LIST:          0 GB         2 KB      0.00%
+
+.. note:: Depending on XIP implementation, this can cause a performance impact in both time and power.
+
+non-XIP Supported external flash
+--------------------------------
+If the external flash does not support XIP, then the firmware patch has to be copied to the internal flash before execution.
```

Comment:
You can choose to not use XIP even if the external flash supports XIP. I suggest that we use "application" instead
```suggestion
If the application does not support XIP from external flash, then the firmware patch has to be copied to the internal flash before execution.
```